src/prerender.ts

```diff
@@ -214,8 +214,12 @@ export async function prerender(nitro: Nitro) {
     const isImplicitHTML =
       !route.endsWith(".html") && contentType.includes("html");
     const routeWithIndex = route.endsWith("/") ? route + "index" : route;
+    const htmlPath =
+      route.endsWith("/") || nitro.options.experimental.prerenderIndexHTML
```

Comment:
More thinking, we can directly introduce it as a `prerender` option as a fix since unblocks platform compatibility. (ideas welcome for naming!)

Reply:
`prerender.forceSubfolder`? As we should render subfolder by default for an explicit trailing slash route, I think.

Reply:
Force has some negative sense to me. Maybe `auto`? (also would be nice that `html` is included in key somehow to clarify behavior)

Reply:
I think the default is true (for using subfolders). how about `prerender.autoSubfolderIndex`?

Reply:
`autoSubfolderIndex` seems nice!